src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -603,15 +612,64 @@ public Kernel parseArgs(String... args) {
         }
 
         // Update device configuration from commandline arguments after loading config files
-        kernelCommandLine.updateDeviceConfiguration(getContext().get(DeviceConfiguration.class));
+        DeviceConfiguration deviceConfiguration = getContext().get(DeviceConfiguration.class);
+        kernelCommandLine.updateDeviceConfiguration(deviceConfiguration);
+        // After configuration is fully loaded, initialize Nucleus service config
+        initializeNucleusServiceConfigIfNotExist(kernelAlts, deviceConfiguration);
+
         setupProxy();
+
         return this;
     }
 
     private void setupProxy() {
         ProxyUtils.setProxyProperties(context.get(DeviceConfiguration.class));
     }
 
+    private void initializeNucleusServiceConfigIfNotExist(KernelAlternatives kernelAlternatives,
```

Comment:
After adding getting the recipe from the build file, will we still keep the properties file as well for version? or should the version now come from the recipe?

Reply:
Ideally properties file can be removed. Let me see if I can get the order of the events correct